[BUG] _CMAKE build failure w/g++-9_

**Summary**

Build fails in kspace.cpp with omp related issue.

**LAMMPS Version and Platform**

version: current git master: daac3f7102f4d12b05d9ec24d13bfdf675496623

machine: mac w/homebrew g++-9 (gcc version 9.1.0 (Homebrew GCC 9.1.0))

cmake version 3.14.4


**Expected Behavior**

build completes

**Actual Behavior**

```
[ 56%] Building CXX object CMakeFiles/lmp.dir/Users/relliott/unison-sync/KIM/git/lammps/src/kspace.cpp.o
/Users/relliott/unison-sync/KIM/git/lammps/src/kspace.cpp: In member function 'void LAMMPS_NS::KSpace::qsum_qsq()':
/Users/relliott/unison-sync/KIM/git/lammps/src/kspace.cpp:278:3: error: 'nlocal' not specified in enclosing 'parallel'
  278 |   for (int i = 0; i < nlocal; i++) {
      |   ^~~
/Users/relliott/unison-sync/KIM/git/lammps/src/kspace.cpp:276:9: error: enclosing 'parallel'
  276 | #pragma omp parallel for default(none) reduction(+:qsum_local,qsqsum_local)
      |         ^~~
/Users/relliott/unison-sync/KIM/git/lammps/src/kspace.cpp:279:19: error: 'q' not specified in enclosing 'parallel'
  279 |     qsum_local += q[i];
      |                   ^
/Users/relliott/unison-sync/KIM/git/lammps/src/kspace.cpp:276:9: error: enclosing 'parallel'
  276 | #pragma omp parallel for default(none) reduction(+:qsum_local,qsqsum_local)
      |         ^~~
make[2]: *** [CMakeFiles/lmp.dir/Users/relliott/unison-sync/KIM/git/lammps/src/kspace.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/lmp.dir/all] Error 2
make: *** [all] Error 2
```


**Steps to Reproduce**

CXX=g++-9 cmake ../cmake/
make

--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -316,7 +316,18 @@ pkg_depends(USER-SCAFACOS MPI)
 
 include(CheckIncludeFileCXX)
 find_package(OpenMP QUIET)
-option(BUILD_OMP "Build with OpenMP support" ${OpenMP_FOUND})
+
+# TODO: this is a temporary workaround until a better solution is found. AK 2019-05-30
+# GNU GCC 9.x uses settings incompatible with our use of 'default(none)' in OpenMP pragmas
+# where we assume older GCC semantics. For the time being, we disable OpenMP by default
+# for GCC 9.x and beyond. People may manually turn it on, but need to run the script
+# src/USER-OMP/hack_openmp_for_pgi_gcc9.sh on all sources to make it compatible with gcc 9.
+if ((${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU") AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 9.0.0))
+  option(BUILD_OMP "Build with OpenMP support" OFF)
+else()
+  option(BUILD_OMP "Build with OpenMP support" ${OpenMP_FOUND})
+endif()
+
 if(BUILD_OMP)
   find_package(OpenMP REQUIRED)
   check_include_file_cxx(omp.h HAVE_OMP_H_INCLUDE)

